components/zcash_protocol/src/consensus.rs

```diff
@@ -286,7 +286,7 @@ pub trait Parameters: Clone {
     /// Determines whether the specified network upgrade is active as of the
     /// provided block height on the network to which this Parameters value applies.
     fn is_nu_active(&self, nu: NetworkUpgrade, height: BlockHeight) -> bool {
-        self.activation_height(nu).map_or(false, |h| h <= height)
+        self.activation_height(nu).is_some_and(|h| h <= height)
```

Comment:
Bug: this was added in Rust 1.70, but the MSRV for this particular crate is Rust 1.52 (because it's so simple, we've left it lower in the past in order to enable it to be used in the widest possible set of environments).

I expect your entire "Fix beta clippy complaints" commit needs to be removed. We certainly can't bump MSRV in 0.6.3 or 0.7.1; the earliest would be 0.8 if we want to bump it at all.